rust/src/storage/s3/mod.rs

```diff
@@ -955,6 +954,9 @@ mod tests {
             s3_storage_options::AWS_WEB_IDENTITY_TOKEN_FILE,
             "token_file",
         );
+        std::env::remove_var(s3_storage_options::AWS_S3_POOL_IDLE_TIMEOUT_SECONDS);
+        std::env::remove_var(s3_storage_options::AWS_STS_POOL_IDLE_TIMEOUT_SECONDS);
+        std::env::remove_var(s3_storage_options::AWS_S3_GET_INTERNAL_SERVER_ERROR_RETRIES);
 
```

Comment:
I was getting random errors, I believe it was caused by unpredictable order of execution since it uses the mutual state, envs. This fixes that